### PR TITLE
Enhance subcategories

### PR DESCRIPTION
**Pull request by tsiegleauq**
- show parents over children in motion list, motion detail and PDF
- indent children of categories in the filter list

### CODE REVIEW

File: client/src/app/shared/components/sort-filter-bar/filter-menu/filter-menu.component.ts [hunk 1]
```diff
@@ -60,6 +60,8 @@ export class FilterMenuComponent implements OnInit {
     }
 
     public isFilter(option: OsFilterOption): boolean {
+        console.log('isFilter: ', option);
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/sort-filter-bar/filter-menu/filter-menu.component.ts]: Please remove :)

File: client/src/app/site/base/base-view-model.ts [hunk 2]
```diff
@@ -45,6 +45,10 @@ export abstract class BaseViewModel<M extends BaseModel = any>
         return `${this.collectionString}:${this.id}`;
     }
 
+    public get parent(): BaseViewModel<M> {
```
Comment (FinnStutzenstein) [inline client/src/app/site/base/base-view-model.ts]: Please remove this. Not every view model does have a parent. Also calling this function (if it is not overwritten) creates an endless recursion.

If you need to have some BaseModels with the property parent, create an interface for this.

File: client/src/app/site/motions/models/view-category.ts [hunk 3]
```diff
@@ -64,6 +64,14 @@ export class ViewCategory extends BaseViewModel<Category> implements CategoryTit
         return name;
     }
 
+    public get nameWithParentAbove(): string {
```
Comment (FinnStutzenstein) [inline client/src/app/site/motions/models/view-category.ts]: Please think of categories with multiple parents in the category-tree. Either document, that only the direct parent is used, or fix this function, that every parent is used (I do not know, if this is intended or not..)

File: client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.html [hunk 4]
```diff
@@ -104,35 +104,6 @@
...
 
-                <!-- state column -->
-                <ng-container matColumnDef="state">
-                    <mat-header-cell *matHeaderCellDef mat-sort-header>State</mat-header-cell>
-                    <mat-cell (click)="openEditInfo(motion, $event)" *matCellDef="let motion">
-                        <div class="fill">
-                            <div class="innerTable state-column">
-                                <div class="ellipsis-overflow" *ngIf="motion.category">
-                                    <os-icon-container icon="device_hub">
-                                        {{ motion.category.prefixedNameWithParents }}
```
Comment (FinnStutzenstein) [inline client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.html]: Did you remove all usages of `prefixedNameWithParents`? If so, please remove this property.

File: client/src/app/site/motions/models/view-category.ts (hunk 3 above)
Comment (tsiegleauq) [reply client/src/app/site/motions/models/view-category.ts]: According to my information, thouse are irrelevant.

File: client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.html (hunk 4 above)
Comment (tsiegleauq) [reply client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.html]: Still used in selector classes

**Pull request merged**